test/e2e/set/scylladbmonitoring/scylladbmonitoring.go

```diff
@@ -0,0 +1,241 @@
+		// Some of these may be fixable by manually verifying it in the operator sync loop so it can also be
+		// consumed by clients, but it's a bigger effort.
+
+		framework.By("Verifying that Prometheus is configured correctly")
+
+		prometheusServingCABundleConfigMap, err := f.KubeClient().CoreV1().ConfigMaps(f.Namespace()).Get(ctx, fmt.Sprintf("%s-prometheus-serving-ca", sc.Name), metav1.GetOptions{})
+		o.Expect(err).NotTo(o.HaveOccurred())
+		prometheusServingCACerts, _ := verification.VerifyAndParseCABundle(prometheusServingCABundleConfigMap)
+		o.Expect(prometheusServingCACerts).To(o.HaveLen(1))
+
+		prometheusServingCAPool := x509.NewCertPool()
+		prometheusServingCAPool.AddCert(prometheusServingCACerts[0])
+
+		prometheusGrafanaClientSecret, err := f.KubeClient().CoreV1().Secrets(f.Namespace()).Get(ctx, fmt.Sprintf("%s-prometheus-client-grafana", sc.Name), metav1.GetOptions{})
+		o.Expect(err).NotTo(o.HaveOccurred())
+		_, prometheusGrafanaClientCertBytes, _, prometheusGrafanaClientKeyBytes := verification.VerifyAndParseTLSCert(prometheusGrafanaClientSecret, verification.TLSCertOptions{
+			IsCA:     pointer.Bool(false),
+			KeyUsage: opointer.KeyUsage(x509.KeyUsageKeyEncipherment | x509.KeyUsageDigitalSignature),
+		})
+
+		prometheusGrafanaAdminTLSCert, err := tls.X509KeyPair(prometheusGrafanaClientCertBytes, prometheusGrafanaClientKeyBytes)
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		o.Expect(sm.Spec.Components.Prometheus.ExposeOptions.WebInterface.Ingress.DNSDomains).To(o.HaveLen(1))
+		prometheusServerName := sm.Spec.Components.Prometheus.ExposeOptions.WebInterface.Ingress.DNSDomains[0]
+
+		promHTTPClient, err := prometheusappclient.NewClient(prometheusappclient.Config{
+			Address: "https://" + f.GetIngressAddress(prometheusServerName),
+			Client: &http.Client{
+				Transport: &http.Transport{
+					TLSClientConfig: &tls.Config{
+						ServerName:   prometheusServerName,
+						Certificates: []tls.Certificate{prometheusGrafanaAdminTLSCert},
+						RootCAs:      prometheusServingCAPool,
+					},
+					Proxy: http.ProxyFromEnvironment,
+					DialContext: (&net.Dialer{
+						Timeout:   30 * time.Second,
+						KeepAlive: 30 * time.Second,
+					}).DialContext,
+					ForceAttemptHTTP2:     true,
+					MaxIdleConns:          100,
+					IdleConnTimeout:       90 * time.Second,
+					TLSHandshakeTimeout:   10 * time.Second,
+					ExpectContinueTimeout: 1 * time.Second,
+				},
+			},
+		})
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		promClient := promeheusappv1api.NewAPI(promHTTPClient)
+
+		o.Eventually(func(eo o.Gomega) {
+			ctxTargets, ctxTargetsCancel := context.WithTimeout(ctx, 15*time.Second)
+			defer ctxTargetsCancel()
+
+			targets, err := promClient.Targets(ctxTargets)
+			framework.Infof("Listing grafana targets: err: %v, active: %d, dropped: %d", err, len(targets.Active), len(targets.Dropped))
+			eo.Expect(err).NotTo(o.HaveOccurred())
+
+			eo.Expect(targets.Active).To(o.HaveLen(2))
```

Comment:
2 because there're that many rules in serviceMonitor? There're three actually, but one of them is not reachable (it's mentioned in other review remark). 
Would be good to explain in the comment, because it depends on the assets.

Reply:
actually we could fetch created serviceMonitors and compare this number with aggregated number of scrape rules.

Reply:
I think that would have been discovered if we didn't have to ignore the dozens od active targets for which there is an issue. I am adding a comment but I am open to extending this in the future in other ways as well.

Reply:
to my understanding, those are mutually exclusive, so enforcing 0 dropped targets should mean all have been processed fine